components/eventing-controller/cmd/eventing-pre-upgrade-job-1-24/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kelseyhightower/envconfig"
+
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	publisherDeployment "github.com/kyma-project/kyma/components/eventing-controller/pkg/deployment"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/clients/deployment"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/clients/eventingbackend"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/clients/eventmesh"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/clients/secret"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/clients/subscription"
+	"github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/env"
+	jobprocess "github.com/kyma-project/kyma/components/eventing-controller/upgrade-job/process"
+)
+
+func main() {
+	// Read Env variables
+	cfg := new(env.Config)
+	if err := envconfig.Process("", cfg); err != nil {
+		log.Fatalf("Start handler failed with error: %s", err)
+	}
+
+	// Create logger instance
+	ctrLogger, err := logger.New(cfg.LogFormat, cfg.LogLevel)
+	if err != nil {
+		log.Fatalf("failed to initialize logger: %s", err)
+	}
+	defer func() {
+		if err := ctrLogger.WithContext().Sync(); err != nil {
+			log.Printf("failed to flush logger:: %s\n", err)
+		}
+	}()
+
+	// Generate dynamic clients
+	k8sConfig := config.GetConfigOrDie()
+
+	// Create dynamic client (k8s)
+	dynamicClient := dynamic.NewForConfigOrDie(k8sConfig)
+
+	// setup clients
+	deploymentClient := deployment.NewClient(dynamicClient)
+	subscriptionClient := subscription.NewClient(dynamicClient)
+	eventingBackendClient := eventingbackend.NewClient(dynamicClient)
+	secretClient := secret.NewClient(dynamicClient)
+	eventMeshClient := eventmesh.NewClient()
+
+	// Create process
+	p := jobprocess.Process{
+		Logger:             ctrLogger.Logger,
+		TimeoutPeriod:      60 * time.Minute,
+		ReleaseName:        cfg.ReleaseName,
+		Domain:             cfg.Domain,
+		KymaNamespace:      cfg.KymaNamespace,
+		ControllerName:     cfg.EventingControllerName,
+		PublisherName:      publisherDeployment.PublisherName,
+		PublisherNamespace: publisherDeployment.PublisherNamespace,
+		Clients: jobprocess.Clients{
+			Deployment:      deploymentClient,
+			Subscription:    subscriptionClient,
+			EventingBackend: eventingBackendClient,
+			Secret:          secretClient,
+			EventMesh:       eventMeshClient,
+		},
+		State: jobprocess.State{},
+	}
+
+	checkClusterVersion := jobprocess.NewCheckClusterVersion(&p)
+	if err := checkClusterVersion.Do(); err != nil {
+		ctrLogger.Logger.WithContext().Error(errors.Wrapf(err, "failed to check: %s", checkClusterVersion.ToString()), p.KymaNamespace)
+		os.Exit(1)
+	}
+
+	checkBebIsEnabled := jobprocess.NewCheckIsBebEnabled(&p)
```

Comment:
I recommend to add both `NewCheckClusterVersion(&p)` and `NewCheckIsBebEnabled(&p)` to the `p.Steps` inside `func (p *Process) AddSteps()` and remove them from the `main func` to have all steps together, instead of calling some steps in `main` and other steps later.

Reply:
I wanted to suggest that too, but it would make skipping the next steps without throwing an error a bit complicated. In other words, you could see being version 1.24 and is BEB enabled, as prerequisites for the migration steps.